src/main/java/uk/ac/ebi/ampt2d/metadata/rest/controllers/ReferenceSequenceRestController.java

```diff
@@ -33,44 +34,46 @@
-public class AssemblyRestController implements ResourceProcessor<RepositoryLinksResource> {
+@Api(tags = "ReferenceSequence Entity")
+@RequestMapping(path = "reference-sequences")
+public class ReferenceSequenceRestController implements ResourceProcessor<RepositoryLinksResource> {
 
     @Autowired
-    private AssemblyRepository assemblyRepository;
+    private ReferenceSequenceRepository referenceSequenceRepository;
 
     @Autowired
-    private GenericResourceAssembler<Assembly, AssemblyResource> resourceAssembler;
+    private GenericResourceAssembler<ReferenceSequence, ReferenceSequenceResource> resourceAssembler;
 
-    @ApiOperation(value="Get a filtered list of assemblies based on filtering criteria")
+    @ApiOperation(value="Get a filtered list of reference sequences based on filtering criteria")
     @ApiImplicitParams({
             @ApiImplicitParam(name = "name", value = "name", dataType = "string", paramType = "query", example = "GRCh38"),
             @ApiImplicitParam(name = "patch", value = "patch number", dataType = "string", paramType = "query", example = "p2"),
-            @ApiImplicitParam(name = "accessions", value = "accession", dataType = "string", paramType = "query", example = "GCA_000001405.3")
+            @ApiImplicitParam(name = "accessions", value = "accession", dataType = "string", paramType = "query", example = "GCA_000001405.3"),
+            @ApiImplicitParam(name = "type", value = "Reference Sequence's type", dataType = "string",
+                    paramType = "query", example = "ASSEMBLY", allowableValues = "ASSEMBLY,GENE,TRANSCRIPTOME")
     })
     @RequestMapping(method = RequestMethod.GET, path = "search", produces = "application/json")
     @ResponseBody
     @SuppressWarnings("unchecked")
-    public ResponseEntity<Resources<AssemblyResource>> search(@QuerydslPredicate(root = Assembly.class) Predicate predicate) {
-        List<Assembly> assemblies = (List<Assembly>) assemblyRepository.findAll(predicate);
+    public ResponseEntity<Resources<ReferenceSequenceResource>> search(@QuerydslPredicate(root = ReferenceSequence.class) Predicate predicate) {
+        List<ReferenceSequence> referenceSequences = (List<ReferenceSequence>) referenceSequenceRepository.findAll(predicate);
 
-        Resources<AssemblyResource> resources = (Resources<AssemblyResource>) resourceAssembler.toResources(Assembly.class, assemblies);
+        Resources<ReferenceSequenceResource> resources = (Resources<ReferenceSequenceResource>) resourceAssembler.toResources(ReferenceSequence.class, referenceSequences);
 
         return ResponseEntity.ok(resources);
     }
 
     @Override
     public RepositoryLinksResource process(RepositoryLinksResource resource) {
-        resource.add(ControllerLinkBuilder.linkTo(AssemblyRestController.class).slash("/search").withRel("assemblies"));
+        resource.add(ControllerLinkBuilder.linkTo(ReferenceSequenceRestController.class).slash("/search").withRel("referenceSequences"));
```

Comment:
Can you also change this to `reference-seequences` please?

Reply:
Done.
Looked another place during testing and working on it